Framework/Algorithms/src/PolarizationCorrections/HeliumAnalyserEfficiency.cpp

```diff
@@ -103,20 +103,15 @@ void HeliumAnalyserEfficiency::init() {
  */
 std::map<std::string, std::string> HeliumAnalyserEfficiency::validateInputs() {
   std::map<std::string, std::string> errorList;
-  const std::string inputWorkspaceName = getProperty(PropertyNames::INPUT_WORKSPACE);
-  if (!AnalysisDataService::Instance().doesExist(inputWorkspaceName)) {
-    errorList[PropertyNames::INPUT_WORKSPACE] =
-        "The input workspace " + inputWorkspaceName + " does not exist in the ADS.";
+  WorkspaceGroup_sptr ws = getProperty(PropertyNames::INPUT_WORKSPACE);
+
+  if (!ws->isGroup()) {
+    errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a group workspace";
   } else {
-    const auto ws = AnalysisDataService::Instance().retrieve(getProperty(PropertyNames::INPUT_WORKSPACE));
-    if (!ws->isGroup()) {
-      errorList[PropertyNames::INPUT_WORKSPACE] = "The input workspace is not a group workspace";
-    } else {
-      const auto wsGroup = std::dynamic_pointer_cast<WorkspaceGroup>(ws);
-      if (wsGroup->size() != 4) {
-        errorList[PropertyNames::INPUT_WORKSPACE] =
-            "The input group workspace must have four periods corresponding to the four spin configurations.";
-      }
+    const auto wsGroup = std::dynamic_pointer_cast<WorkspaceGroup>(ws);
+    if (wsGroup->size() != 4) {
```

Comment:
Now that we're getting the workspace group directly from the property on line 106, this can be streamlined to remove the dynamic_pointer_cast (i.e. remove line 111) and just have `if(ws->size() != 4)`. To improve readability of the code, we might like to name the original variable on line 106 `wsGroup` so that we can continue to use that name here in the `if` statement and it's clear to future developers that this variable holds a workspace group.